src/main/java/org/zone/region/flag/entity/player/move/message/leaving/LeavingFlag.java

```diff
@@ -0,0 +1,47 @@
+package org.zone.region.flag.entity.player.move.message.leaving;
+
+import net.kyori.adventure.text.Component;
+import org.jetbrains.annotations.NotNull;
+import org.zone.region.flag.Flag;
+import org.zone.region.flag.FlagTypes;
+import org.zone.region.flag.entity.player.move.message.display.MessageDisplay;
+import org.zone.region.flag.entity.player.move.message.display.MessageDisplayTypes;
+
+public class LeavingFlag implements Flag {
+
+    private @NotNull MessageDisplay messageDisplay = MessageDisplayTypes.CHAT_MESSAGE_DISPLAY.createCopyOfDefault();
+    private @NotNull Component leavingMessage;
+    @Deprecated(forRemoval = true)
+    private @NotNull Component oldLeavingMessage;
+
+    public LeavingFlag(@NotNull Component leavingMessage, @NotNull MessageDisplay messageDisplay) {
+        this.leavingMessage = leavingMessage;
+        this.messageDisplay = messageDisplay;
+    }
+
+    @Deprecated(forRemoval = true)
+    public LeavingFlag(@NotNull Component leavingMessage) {
+        this.oldLeavingMessage = leavingMessage;
```

Comment:
``leavingMessage`` is ``@NotNull`` yet your leaving it null, this will cause errors